components/tenant-mgt/org.wso2.carbon.tenant.mgt/src/main/java/org/wso2/carbon/tenant/mgt/util/TenantMgtUtil.java

```diff
@@ -523,14 +562,16 @@ public static void addAdditionalClaimsToUserStoreManager(Tenant tenant) throws E
             }
 
             // Can be extended to store other user information.
-            UserStoreManager userStoreManager =
-                    (UserStoreManager) TenantMgtServiceComponent.getRealmService().
-                            getTenantUserRealm(tenant.getId()).getUserStoreManager();
+            UserStoreManager userStoreManager = getUserStoreManager(tenant.getId());
             if (!userStoreManager.isReadOnly()) {
                 userStoreManager.setUserClaimValues(tenant.getAdminName(), claimsMap,
                         UserCoreConstants.DEFAULT_PROFILE);
             }
         } catch (org.wso2.carbon.user.api.UserStoreException e) {
+            if (e instanceof UserStoreClientException) {
```

Comment:
Can't we catch UserStoreClientException specifically instead of having instanceof check ?

Reply:
Fixed in https://github.com/wso2/carbon-multitenancy/pull/276/commits/3fd11ef5ee2c9ee187a4ec951bb8b5eb59acd6e1